### PR TITLE
chore: adding `MsgChannelUpgradeCancel` fields and `sdk.Msg` interface impl

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding fields to `MsgChannelUpgradeCancel`
- Implement `ValidateBasic` and `GetSigners` methods for `sdk.Msg` interface
- Adding tests

closes: #1624 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/core/04-channel/types/msgs.go [hunk 1]
```diff
@@ -589,16 +589,59 @@ var _ sdk.Msg = &MsgChannelUpgradeCancel{}
...
+
+	if len(msg.ProofErrorReceipt) == 0 {
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty proof")
+	}
+
+	if msg.ProofHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "proof height must be non-zero")
+	}
+
+	if msg.ErrorReceipt.Sequence == 0 {
```
Comment (chatton) [inline modules/core/04-channel/types/msgs.go:627]: [q] maybe this isn't possible, but should this be `<= 0`, a negative value is currently valid. The same applies for `ProofErrorReceipt`

Comment (damiannolan) [reply modules/core/04-channel/types/msgs.go:627]: The sequence field is a unsigned int as far as I know

Comment (chatton) [reply modules/core/04-channel/types/msgs.go:627]: ah perfect, not an issue then 👍

**Review by chatton — APPROVED** [review]
LGTM! Nice test coverage 🥇

**Review by crodriguezvega — APPROVED** [review]
LGTM! 👍

**Pull request merged**